### DISCUSSION TITLE
OpenAI Provider

### DISCUSSION THREAD

**Discussion by monksy** [mealie-recipes/mealie#4073] [General]
Is there a model or what would we need to use Ollama instead of OpenAI? I'm pretty happy to see we're getting the image recognition feature here. I hope there will be an effort to go to more of an open source version of that.

**Comment by michael-genson** [comment]
I'm not _too_ familiar with Ollama, but they have OpenAI compatible endpoints: https://ollama.com/blog/openai-compatibility

I haven't tested this out at all, unsure of image recognition capability. You can change your OpenAI base URL and API key using env vars (for Ollama specifically the API key is ignored, but you still must pass it). More info in the docs: https://docs.mealie.io/documentation/getting-started/installation/backend-config/#openai

**Reply by monksy** [reply]
I was just opening up the conversation here for ollama compatiblity here. I didn't know if there were special requests in Open AI for image rec.

**Reply by death2all110** [reply]
Seems image recognition is not working with Ollama, at least for me. Not sure if there are other self-hosted AI with OpenAI API support.

https://github.com/mealie-recipes/mealie/issues/4097

**Reply by death2all110** [reply]
After a bunch of troubleshooting, and debugging, it looks like the image format is the issue. Mealie currently passes the image as a webp image, which is not currently supported by ollama. Currently only png/jpg are supported. https://github.com/ollama/ollama/issues/2457

**Reply by michael-genson** [reply]
FYI: https://github.com/mealie-recipes/mealie/pull/4117

Hopefully this fixes it

**Reply by monksy** [reply]
Seeing that and the fact that others have jumped up to ask what about ollama makes me so happy. 

I was pissed to find out that OpenAI was demanding non-voip numbers.

**Reply by death2all110** [reply]
Michael is a rockstar.

**Reply by monksy** [reply]
So that's who Chad Kruger was talking about. https://www.youtube.com/watch?v=DmeUuoxyt_E

**Reply by mavproductions** [reply]
> Michael is a rockstar.

Did you get this working w/o issue? 😯
